src/aiida_quantumespresso/calculations/pw.py

```diff
@@ -163,31 +163,25 @@ def validate_inputs_base(value, _):
         parameters = value['parameters'].get_dict()
         calculation_type = parameters.get('CONTROL', {}).get('calculation', 'scf')
 
-        # Check that the restart input parameters are set correctly
-        if calculation_type in ('nscf', 'bands'):
-            if parameters.get('ELECTRONS', {}).get('startingpot', 'file') != 'file':
-                return f'`startingpot` should be set to `file` for a `{calculation_type}` calculation.'
-            if parameters.get('CONTROL', {}).get('restart_mode', 'from_scratch') != 'from_scratch':
-                warnings.warn(f'`restart_mode` should be set to `from_scratch` for a `{calculation_type}` calculation.')
-        elif 'parent_folder' in value:
+        # If a `parent_folder` input is provided, make sure the inputs are set to restart
+        if 'parent_folder' in value and calculation_type not in ('nscf', 'bands'):
```

Comment:
Shouldn't an `scf` calculation also not have "some" restart parameters if a `parent_folder` is specified? Why only `nscf` and `bands`? What is the use case for restarting an `scf` calc `from_scratch` and without reading the CD from file?

Reply:
Got confused myself here again. Note the `not in ('nscf', 'bands')`, i.e. this check is only done for _other_ calculation types. The reason is that for `nscf` and `bands`, the default of `ELECTRONS.startingpot` changes to `'file'`, see:

https://www.quantum-espresso.org/Doc/INPUT_PW.html#idm899

Reply:
Still not sure about the logic though. Why would there only have to be validation for anything but `nscf` and `bands`? For example, if the user specifies `bands` but then `ELECTRONS.startingpot = 'atomic'`, surely that should also emit a warning then?

Reply:
Sure, that'd be another validation we can do. Much less likely use case though, someone what assigns a parent folder and then specifically sets `ELECTRONS.startingpot = 'atomic'` might be up to some weird business that actually requires this input.